Makefile

```diff
@@ -11,3 +11,11 @@ testacc:
 
 install:
 	go build ${BUILD_FLAGS} -o ~/.terraform.d/plugins/terraform-provider-secrethub
+
+GOLANGCI_VERSION=v1.23.8
+
+lint-tools:
+	@curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(go env GOPATH)/bin ${GOLANGCI_VERSION}
+
+lint:
+	@golangci-lint run
```

Comment:
Consider merging the `lint` and `lint-tools` jobs into a single `lint` job:

```
docker run -v $(pwd):/app -w /app golangci/golangci-lint:v1.23.8-alpine golangci-lint run
```

This has the following advantages:
- You don't need to run `make lint-tools` before running `make lint` for the first time (easier to get started with contributing).
- The linter version used is always up-to-date with what is specified in the `Makefile`. With the two separate jobs, you have to know the linter is outdated and update it separately by running `make lint-tools`.
- You can use different linter versions for different projects. Right now, we require to have the linter version as specified to be installed on the workstation in the global path.
- The same environment is used locally as in CI.

But may have the following disadvantages:
- Does `pwd` work on Windows?
- You need to have Docker intsalled to use the command provided by `make lint`. You can of course still install the linter directly on your workstation if you prefer to use without Docker.

I realize we use the same mechanism of two separate jobs in the [CLI](github.com/secrethub/secrethub-cli) and [GO SDK](github.com/secrethub/secrethub-go) projects. I think we should consider changing those as well, for the same reasons.

Reply:
`pwd` does not work on Windows

Reply:
But otherwise its a nice idea. I'm not sure if it not working on windows is a significant issue that we could or should address. What do you think?

Reply:
We do support using the provider on Windows. So anything that's directly used by our users should support Windows.

For development, I think we should strive to support Windows as much as possible, so that every user can also contribute. But given that we haven't had any discussion about this yet, before making any permanent decision about this, let's discuss broader than just the two of us.

For this specific issue, I think we can make it work cross-platform: https://stackoverflow.com/a/41489151

Reply:
The expectation with a `Makefile` is generally that it runs on a Unix system, or at least non-Windows.

Reply:
I got the command to work with docker, but it takes significantly longer (more than 1 minute, almost 2)

Reply:
Okay, I found a way to make it lint in about 2 seconds.